src/hooks/cakeVault/useGetVaultFees.ts

```diff
@@ -0,0 +1,74 @@
+import { useEffect, useState } from 'react'
+import { useCakeVaultContract } from 'hooks/useContract'
+
+export enum FeeFunctions {
+  performanceFee = 'performanceFee',
+  callFee = 'callFee',
+  withdrawalFee = 'withdrawalFee',
+  withdrawalFeePeriod = 'withdrawalFeePeriod',
+}
+
+export interface VaultFees {
+  performanceFee: string
+  callFee: string
+  withdrawalFee: string
+  withdrawalFeePeriod: string
+}
+
+const useGetVaultFees = (functionsArray: FeeFunctions[]) => {
+  const cakeVaultContract = useCakeVaultContract()
+  const [fees, setFees] = useState({
+    performanceFee: null,
+    callFee: null,
+    withdrawalFee: null,
+    withdrawalFeePeriod: null,
+  })
+
+  useEffect(() => {
+    const getPerformanceFee = async () => {
+      const contractPerformanceFee = await cakeVaultContract.methods.performanceFee().call()
+      setFees((prevState) => {
+        return { ...prevState, performanceFee: contractPerformanceFee }
+      })
+    }
+
+    const getCallFee = async () => {
+      const contractCallFee = await cakeVaultContract.methods.callFee().call()
+      setFees((prevState) => {
+        return { ...prevState, callFee: contractCallFee }
+      })
+    }
+
+    const getWithdrawalFee = async () => {
+      const contractWithdrawalFee = await cakeVaultContract.methods.withdrawFee().call()
+      setFees((prevState) => {
+        return { ...prevState, withdrawalFee: contractWithdrawalFee }
+      })
+    }
+
+    const getWithdrawalFeePeriod = async () => {
+      const contractWithdrawalFeeTimePeriod = await cakeVaultContract.methods.withdrawFeePeriod().call()
+      setFees((prevState) => {
+        return { ...prevState, withdrawalFeePeriod: contractWithdrawalFeeTimePeriod }
+      })
+    }
```

Comment:
These should definitely be batched.

Reply:
My _idea_ is that they can be called selectively. 

like here: https://github.com/pancakeswap/pancake-frontend/pull/1007/files#diff-80c27d122268c36a58ec1320fde37aaf56dcf8085b3a08450a164d04dc2d1ef7R27

`const { callFee } = useGetVaultFees([FeeFunctions.callFee])`

Would that impact the use of `makeBatchRequest`?

Separately, I had an issue where passing the functionsArray as a dependency created an infinite loop ([ln 67 - 69](https://github.com/pancakeswap/pancake-frontend/pull/1007/files#diff-ce17aa16fd3da44cb39ee0bf5c238fac3f55368fe1f62c59d0256d341b507e57R67-R68)). 

Still figuring out how best to manage this 'only make the contract calls you need' pattern.

Reply:
here you do 4 node requests on each effect run, it's really inefficient. A batch request will send all the 4 calls in 1 request

Reply:
You could both batch and select which one to call if you don't want all of them all the time.
```
// prepare the calls
...
const feeCalls = {
      performanceFee: getPerformanceFeeCall,
      callFee: getCallFeeCall,
      withdrawalFee: getWithdrawalFeeCall,
      withdrawalFeePeriod: getWithdrawalFeePeriodCall,
    }
const calls = functionsArray.map((functionName) => feeCalls[functionName])
makeBatchRequest(calls)
```
If the call is not in the array it just doesn't get mapped and your batching 3 calls instead of 4 if needed.

As for array as dependency - might need to google that up a bit, its annoying yeah...

OR you can just redefine the parameter as object `calls: FeeCalls` and pass something like
```
{
  performanceFee: false
  callFee: true
  withdrawalFee: true
  withdrawalFeePeriod: true
}
```

and in deps array
```
const { performanceFee, callFee, withdrawalFee, withdrawalFeePeriod } = calls;
useEffect(() => {
  ...
}, [performanceFee, callFee, withdrawalFee, withdrawalFeePeriod ])
```

then you can extract convert it to array to map for batching stuff above.

Reply:
Batched and simplified - thanks Cheems but out of time for anything else